llvm/lib/ObjectYAML/DXContainerYAML.cpp

```diff
@@ -188,6 +208,13 @@ void MappingTraits<DXContainerYAML::Signature>::mapping(
   IO.mapRequired("Parameters", S.Parameters);
 }
 
+void MappingTraits<DXContainerYAML::RootSignatureDesc>::mapping(
+    IO &IO, DXContainerYAML::RootSignatureDesc &S) {
+  IO.mapRequired("Version", S.Version);
+#define ROOT_ELEMENT_FLAG(Num, Val, Str) IO.mapRequired(#Val, S.Val);
```

Comment:
We could instead consider doing this as a `mapOptional` with `false` as the default value. That would make it more concise in the YAML representation.